src/main_thread/init/utils/update_manifest_codec_support.ts

```diff
@@ -0,0 +1,155 @@
+import isCodecSupported from "../../../compat/is_codec_supported";
+import { MediaError } from "../../../errors";
+import type { ICodecSupportList, IManifestMetadata } from "../../../manifest";
+import cdmCodecSupportProber from "../../../mse/cdm_codec_support_prober";
+import type { ITrackType } from "../../../public_types";
+/**
+ * Ensure that all `Representation` and `Adaptation` have a known status
+ * for their codec support and probe it for cases where that's not the
+ * case.
+ *
+ * Because probing for codec support is always synchronous in the main thread,
+ * calling this function ensures that support is now known.
+ *
+ * @param {Object} manifest
+ */
+export function updateManifestCodecSupport(
+  manifest: IManifestMetadata,
+): ICodecSupportList {
+  const codecSupportList: ICodecSupportList = [];
+  const codecSupportedByMseMap: Map<string, Map<string, boolean>> = new Map();
+  const codecSupportedByCdmMap: Map<string, Map<string, boolean>> = new Map();
+
+  manifest.periods.forEach((p) => {
+    [
+      ...(p.adaptations.audio ?? []),
+      ...(p.adaptations.video ?? []),
+      ...(p.adaptations.text ?? []),
+    ].forEach((a) => {
+      let hasSupportedCodecs = false;
+      a.representations.forEach((r) => {
+        const isEncrypted = r.contentProtections !== undefined;
```

Comment:
Is there a possibility that checking that `representation.contentProtection` is undefined is not enough to determine if content is encrypted? 
For exemple a contentProtection with zero filled keyIds , (related to https://github.com/canalplus/rx-player/pull/1466)